src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -273,73 +275,43 @@ internal static class StreamHelper
 
         #region Static Methods
 
-        internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet)
+        internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet, CancellationTokenSource cts)
         {
-            byte[] data = new byte[ChunkSize];
-
-            int read = 0;
-            long totalWritten = 0;
-            do
+            if (cmdlet == null)
             {
-                if (cmdlet != null)
-                {
-                    ProgressRecord record = new ProgressRecord(ActivityId,
-                        WebCmdletStrings.WriteRequestProgressActivity,
-                        StringUtil.Format(WebCmdletStrings.WriteRequestProgressStatus, totalWritten));
-                    cmdlet.WriteProgress(record);
-                }
-
-                read = input.Read(data, 0, ChunkSize);
+                throw new ArgumentNullException(nameof(cmdlet));
+            }
 
-                if (0 < read)
-                {
-                    output.Write(data, 0, read);
-                    totalWritten += read;
-                }
-            } while (read != 0);
+            Task copyTask = input.CopyToAsync(output, cts.Token);
 
-            if (cmdlet != null)
+            ProgressRecord record = new ProgressRecord(
+                ActivityId,
+                WebCmdletStrings.WriteRequestProgressActivity,
+                StringUtil.Format(WebCmdletStrings.WriteRequestComplete, 0));
+            do
             {
-                ProgressRecord record = new ProgressRecord(ActivityId,
-                    WebCmdletStrings.WriteRequestProgressActivity,
-                    StringUtil.Format(WebCmdletStrings.WriteRequestComplete, totalWritten));
-                record.RecordType = ProgressRecordType.Completed;
+                record.StatusDescription = StringUtil.Format(WebCmdletStrings.WriteRequestComplete, output.Position);
```

Comment:
I think you should use `WebCmdletStrings.WriteRequestProgressStatus` instead, no?

Reply:
Yes. fixed.